Add config properties for HTTP security headers (#97158)
elastic/kibana: docs/setup/settings.asciidoc

proxy sitting in front of it. This determines whether HTTP compression may be used for responses, based on the request `Referer` header.
This setting may not be used when <<server-compression, `server.compression.enabled`>> is set to `false`. *Default: `none`*

a| [[server-securityResponseHeaders-strictTransportSecurity]]
----
server.securityResponseHeaders:
  strictTransportSecurity:
----
| Controls whether the https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Strict-Transport-Security[`Strict-Transport-Security`]
header is used in all responses to the client from the {kib} server. Set this to `null` to disable it. *Default:* enabled

a|
----
server.securityResponseHeaders:
  strictTransportSecurity.maxAge:
----
| Specifies the `max-age` directive of the
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Strict-Transport-Security[`Strict-Transport-Security`] response header. This is

> rounded down to the nearest second. *Default:* `'1Y'` (31536000 seconds)

Reviewer: As discussed, the settings here are a bit inconsistent right now with the transport policy header being controlled using individual settings whereas permissions policy having to be set manually as the raw header value.

Would be good to make that consistent. Maybe the easiest thing would be to let users set the raw header value themselves so that we don't have to keep updating it manually when the spec changes.

Author: Agreed, done in c358373!

2+a|
[TIP]
============
Use a string of `<count>[s\|m\|h\|d\|w\|M\|Y]` (e.g. '20m', '24h', '120d', '1Y').
============

a|
----
server.securityResponseHeaders:
  strictTransportSecurity.includeSubDomains:
----
| Specifies the `includeSubDomains` directive of the
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Strict-Transport-Security[`Strict-Transport-Security`] response header. If
enabled, the directive is included in the header. *Default:* `false`

a|
----
server.securityResponseHeaders:

>   strictTransportSecurity.preload:

Reviewer: @elastic/kibana-docs  What's the best way of printing long settings like this? The table is too narrow to fit both the name and description. However, breaking it up like this makes it a bit confusing for users too.

Reviewer 2: @thomheymann the table capabilities in Asciidoc are limited, and I agree that breaking up the setting names is confusing. These would be best displayed in a definition list instead of a table. I propose that we merge these changes, then I'll fix the formatting before 7.13 releases.

----
| Controls whether the https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Content-Type-Options[`X-Content-Type-Options`] header is
used in all responses to the client from the {kib} server, and specifies what value is used. Allowed values are `nosniff` or `null`. Set
this to `null` to disable it. *Default:* `"nosniff"`

a| [[server-securityResponseHeaders-referrerPolicy]]
----
server.securityResponseHeaders:
  referrerPolicy:
----
| Controls whether the https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Referrer-Policy[`Referrer-Policy`] header is used in all
responses to the client from the {kib} server, and specifies what value is used. Allowed values are `no-referrer`,
`no-referrer-when-downgrade`, `origin`, `origin-when-cross-origin`, `same-origin`, `strict-origin`, `strict-origin-when-cross-origin`,
`unsafe-url`, or `null`. Set this to `null` to disable it. *Default:* `"no-referrer-when-downgrade"`

a| [[server-securityResponseHeaders-permissionsPolicy]]
----
server.securityResponseHeaders:
  permissionsPolicy:
----
| Controls whether the https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Feature-Policy[`Permissions-Policy`] header is used in all

> responses to the client from the {kib} server, and specifies what value is used. Allowed values are any text value or `null`. Set this to

Reviewer 2: ```suggestion
responses to the client from the {kib} server, and specifies what value is used. Allowed values are any text value or `null`.
```

`null` to disable it. *Default:* `"camera=(), microphone=()"` (disables the camera and microphone APIs in the browser)

a| [[server-securityResponseHeaders-disableEmbedding]]
----
server.securityResponseHeaders:
  disableEmbedding:
----
| Controls whether the https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy[`Content-Security-Policy`] and

> https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Frame-Options[`X-Frame-Options`] headers will be configured to disable embedding

Reviewer 2: ```suggestion
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Frame-Options[`X-Frame-Options`] headers are configured to disable embedding
```

> {kib} in other webpages using iframes. Set this to `true` to use secure headers that disable embedding; this will add the

Reviewer 2: ```suggestion
{kib} in other webpages using iframes. When set to `true`, secure headers are used to disable embedding, which adds the
```

> `frame-ancestors: 'self'` directive to the `Content-Security-Policy` response header (if you are using the default CSP rules), and it will

Reviewer 2: ```suggestion
`frame-ancestors: 'self'` directive to the `Content-Security-Policy` response header (if you are using the default CSP rules), and adds
```

> add the `X-Frame-Options: SAMEORIGIN` response header as well. *Default:* `false`

Reviewer 2: ```suggestion
the `X-Frame-Options: SAMEORIGIN` response header. *Default:* `false`
```

the <<server-socketTimeout, `server.socketTimeout`>> counter. *Default: `"120000"`*

|[[server-maxPayloadBytes]] `server.maxPayloadBytes:`

>  | deprecated:[7.13.0,"This setting will no longer be supported as of 8.0."]

Reviewer 2: ```suggestion
 | deprecated:[7.13.0,"In 8.0 and later, this setting will no longer be supported."]
```

Author: Thanks! There are a couple of other config properties with the same string (`cpu.cgroup.path.override` and `cpuacct.cgroup.path.override`), I'll update those too.
